xicam/gui/widgets/previewwidget.py

```diff
@@ -39,34 +39,45 @@ def __init__(self):
     def sizeHint(self):
         return QSize(250, 250)
 
+    @threads.method(threadkey='preview', showBusy=False)
     def preview(self, data):
         if isinstance(data, NonDBHeader):
             self.preview_header(data)
         else:
             self.preview_catalog(data)
 
+    @staticmethod
```

Comment:
Catalog Viewer contains logic to guess at what might be an image field, too. Is this possibly the time to start building up a code base of utilities for dealing with BlueskyRuns?

FWIW, the logic over in CatalogView is >1 and <5 and looks at the shape field: https://github.com/Xi-CAM/Xi-cam.CatalogViewer/blob/33e5e7454664c6b46e230e89d59c872045800944/xicam/catalog_viewer/__init__.py#L63

Reply:
Yes, I'm doing the same thing. Its a copy-paste from that PR. I'll move those methods to `xicam.core.data`.